### PR TITLE
[lua] Adjust when TP is gained during WS and remove H2H from "offhand extra hit logic", fix various h2h hit counts

### LINKED ISSUE #9064: 🐛 Incorrect hit count for various hand-to-hand weaponskills
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## Branch affected by issue 

`base`

## Steps to reproduce

Unfortunately, the easiest way to verify requires inserting a print. Put in a print for `hitsDone` towards the end of `xi.weaponskills.calculateRawWSDmg` function (I put it here: <https://github.com/LandSandBoat/server/blob/base/scripts/globals/weaponskills.lua#L610>), then use various hand-to-hand weaponskills. Make sure that neither your support job nor equipment or food or other modifiers grants double attack or anything of the sort, to prevent pollution of the output.

## Expected behavior

Hand-to-hand weaponskills have one more hit than what is assigned for their `params.numHits` in their individual weaponskill scripts, due to the nature of the hand-to-hand skill/weapon. This is rather confusing, and seems to have led to some weaponskills granting extra hits. Every h2h weaponskill that has "twofold" for hits in their in-client description is set to 1 hit (ie. gets 2 hits), except for Final Heaven which is set to 2, granting it 3. Every other h2h weaponskill has their numHits set to what the description says, granting them an additional hit, although Asuran Fists runs into the 8 hit cap so no extra hit is granted to it. Example, using a print as described above: 

Image

Something to consider when addressing this is to determine what value for `params.numHits` should be listed in the WS scripts. It makes sense to list the actual amount, but currently to get the correct amount of hits you would have to enter the actual amount -1.

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

hopefully closes #9064 

tl;dr: we were doing jank with TP gain when damage was non-zero, and h2h was also doing strange things so this PR corrects all of it. H2H just has a single edge case in the main hits TP loop now.

Also adds stp modifier to the 10 tp from extra hits
## Steps to test these changes
use H2H WS, get TP per fist*2+(num hits above the first two hits * 10 * stp) TP return
Also spot check DW, single wield, 2h. E

### CODE REVIEW

File: scripts/globals/weaponskills.lua [hunk 1]
```diff
@@ -423,9 +426,11 @@ xi.weaponskills.calculateRawWSDmg = function(attacker, target, wsID, tp, action,
     end
 
     local offhandSkill = attacker:getWeaponSkillType(xi.slot.SUB)
+    local isH2H        = false
     if calcParams.skillType == xi.skill.HAND_TO_HAND then
         offhandSkill = xi.skill.HAND_TO_HAND
         subTPGain    = mainhandTPGain
+        isH2H        = true
```
Comment (Xaver-DaRed) [inline scripts/globals/weaponskills.lua:433]: is `isH2H` trully needed?
The variable is only used once and we already have access there to `calcParams` so it could be just replaced with `calcParams.skillType == xi.skill.HAND_TO_HAND` directly

**Review by Xaver-DaRed — APPROVED** [review]

File: scripts/globals/weaponskills.lua (hunk 1 above)
Comment (WinterSolstice8) [reply scripts/globals/weaponskills.lua:433]: I didn't want a multi line conditional. that's all. no other reason
